src/extensionFunctions.ts

```diff
@@ -599,4 +601,46 @@ export namespace extensionFunctions {
 			console.log('[' + msg + ']');
 		}
 	}
+
+	export async function validateCommandIDsInSelectedFile(didactUri: vscode.Uri) : Promise<void> {
+		if (didactUri) {
+			await vscode.commands.executeCommand(START_DIDACT_COMMAND, didactUri);
+		}
+		if (DidactWebviewPanel.currentPanel) {
+			const commands : any[] = extensionFunctions.gatherAllCommandsLinks();
+			if (commands && commands.length > 0) {
+				sendTextToOutputChannel(`Starting validation...`);
+				let allOk = true;
+				for(let command of commands) {
+					// validate all commands we found
+					const parsedUrl = url.parse(command, true);
+					const query = parsedUrl.query;
+					if (query.commandId) {
+						const commandId = utils.getValue(query.commandId);
+						if (commandId) {
+							const vsCommands : string[] = await vscode.commands.getCommands(true);
+							var filteredList : string[] = vsCommands.filter( function (command) {
```

Comment:
I think it's more useful to show them all than stop at the first one. After all, validating code doesn't stop at one - it shows you all the validation errors.

Reply:
unless I misunderstand the code, it won't stop validating the command ids available in the didact file, it will stop searching if there is another commandid with the same id in the list of VS Code Command ids.

Reply:
the way the command API works is it will always take the first one with that ID, so if there are multiples with a particular ID, we're kind of hosed anyway

Reply:
This is not important in this algorithm if there are others similar ids or not. What is important is that it stops searching for another one in case there is already one that is found.

Reply:
I'm not sure I understand the distinction. Can you elaborate? Are you saying if they use a single command multiple times, we only need to validate it once? Though that's true, we're splitting hairs on time in the MS here.

Reply:
No, this is not what I mean. (even if you're raising a good points and using a set instead of a list would save cycles)

the result of the filter, the `filteredList` is used only to check if it superior or equal to 1. It means that we are interested only in checking if we have at least one matching element. It means that as soon as one is found, there is no need to check for the other elements in the list.
using `findIndex` or `find` allows to stop searching. Although the `filter` continue to iterate through the list even if one has already been found.

Reply:
I'd think the difference between these 2 alternatives are a few millis at most maybe but in fact `find` is the better way.

Reply:
For the ms involved, I'm going to leave it for now -- but we can revisit this later if speed becomes an issue.

Reply:
not important for this PR as we are below 100ms for 50 commands in the file.
But important to remind when using `filter` in general

for information, it allows to go from time oscillating between 80 and 100ms to drop to a range of 30 to 50ms.

created https://github.com/redhat-developer/vscode-didact/issues/123